File: go-client-sdk/pkg/models/operations/runproject.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-sdks/pkg/models/shared"
)

type RunProjectPathParams struct {
	ProjectID string `pathParam:"style=simple,explode=false,name=projectId"`
}

type RunProjectRequest struct {
	PathParams RunProjectPathParams
	Request    *shared.RunProjectRequestBody `request:"mediaType=application/json"`
}

type RunProjectResponse struct {
	ContentType                       string
	ProjectRunResponsePayload         *shared.ProjectRunResponsePayload
	RunProject422ApplicationJSONAnyOf *interface{}
	StatusCode                        int
	TsoaErrorResponsePayload          *shared.TsoaErrorResponsePayload
}

File: typescript-client-sdk/src/sdk/models/operations/runproject.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class RunProjectPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: string;
}

export class RunProjectRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: RunProjectPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.RunProjectRequestBody;
}

export class RunProjectResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  projectRunResponsePayload?: shared.ProjectRunResponsePayload;

  @SpeakeasyMetadata()
  runProject422ApplicationJSONAnyOf?: any;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  tsoaErrorResponsePayload?: shared.TsoaErrorResponsePayload;
}